### PR TITLE
Merge insertAll and updateAll fail in case of dotted columns

### LINKED ISSUE #233: Merge issue when column name contains dot
The updateAll and insertAll looks not to sanityse column names.

as a result a column named "my.column" will fail if it is not properly quoted as "`my.column`".
in my context the dag before merging does joins on my.column and I found out quoting the column within updateExpr solve the problem

happy to contribute on a pull request if needed

### PR DESCRIPTION
**Pull request by parisni**
This resolves #233 and #208.

This PR add a test to track the issue and also proposes a fix by quoting with backticks the columns coming from the target plan.

### CODE REVIEW

**Comment by databricks-cla-assistant** [comment]
[CLA assistant check](https://cla.dev.databricks.com/delta-io/delta?pullRequest=235) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.databricks.com/delta-io/delta?pullRequest=235) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dev.databricks.com/check/delta-io/delta?pullRequest=235) it.</sub>

**Comment by tdas** [comment]
Thank you for fixing this. This looks good to me!

File: src/main/scala/org/apache/spark/sql/catalyst/plans/logical/merge.scala [hunk 1]
```diff
@@ -266,7 +266,7 @@ object MergeInto {
             // plan.
             fakeTargetPlan.output.map(_.name).map { tgtColName =>
               val resolvedExpr =
-                resolveOrFail(UnresolvedAttribute(tgtColName), fakeSourcePlan, s"$typ clause")
+                resolveOrFail(functions.expr(s"`$tgtColName`").expr, fakeSourcePlan, s"$typ clause")
```
Comment (tdas) [inline src/main/scala/org/apache/spark/sql/catalyst/plans/logical/merge.scala]: hey @parisni while merging this, I made a tiny change. I replaced this with `UnresolvedAttribute.quotedString(s"`$tgtColName`")` to keep it consistent with other code in this file which uses only spark-catalyst (`functions` stuff is spark-sql).

Comment (parisni) [reply src/main/scala/org/apache/spark/sql/catalyst/plans/logical/merge.scala]: @tdas yeah I wasn´t very clear with the unresolvedAttribute stuff. Thanks

**Pull request closed without merging**